### ISSUE TITLE
Server crash if cookie is not correct 

### ISSUE THREAD

**Issue by patrickelectric** [sile/rustun#9]
From: https://github.com/sile/stun_codec/issues/6

**Comment by joaoantoniocardoso** [comment]
To add more information, it happened repeatedly when trying to use it with [this signaller](https://github.com/centricular/webrtcsink/blob/main/signalling/simple-server.py) and [this js](https://github.com/centricular/webrtcsink/blob/main/www/webrtc.js).

And here are a log, if useful:

```
     Running `target/debug/examples/turnsrv --username test --password test123 --server '127.0.0.1:3478'`
Error: Other (cause; assertion failed: `left == right`; assertion failed: `(left == right)` (left: `2994559686`, right: `554869826`): Unexpected MAGIC_COOKIE: actual=0xb27d5ac6, expected=0x2112a442)
HISTORY:
  [0] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.13/src/message.rs:266
  [1] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.13/src/message.rs:295 -- message_type=Type { class: Request, method: Method(1) }, message_len=0, transaction_id=TransactionId(0x5BF53C5D87394399039928FA)
  [2] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.15/src/combinator.rs:1368
  [3] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.13/src/message.rs:430
  [4] at src/turn_message.rs:32
  [5] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.15/src/decode.rs:448
  [6] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/udp.rs:199 -- peer=192.168.0.11:52840
  [7] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/error.rs:19 -- original_error_kind=InvalidInput
  [8] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/share.rs:48
  [9] at src/transport/stun.rs:47
  [10] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.3.7/src/transport/udp.rs:358
  [11] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.3.7/src/transport/udp.rs:193
  [12] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.3.7/src/channel.rs:176
  [13] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.3.7/src/error.rs:61 -- original_error_kind=CodecError
  [14] at src/server/core.rs:426
  [15] at src/server/mod.rs:53
  [16] at examples/turnsrv.rs:51

```

Thank you

**Comment by sile** [comment]
@patrickelectric I fixed this issue in https://github.com/sile/rusturn/pull/5 and just released [rusturn-v0.0.4](https://github.com/sile/rusturn/releases/tag/0.0.4). Thank you for reporting this problem.

@joaoantoniocardoso Thank you for your information. But, I didn't know how to run the signaller and js, so I didn't confirm the behavior.

**Comment by joaoantoniocardoso** [comment]
@sile Thank you, I can confirm that the problem was solved: now it shows a warning and no longer crashes.

```
[2022-03-21T15:23:00Z WARN  rusturn::server] Other (cause; assertion failed: `left == right`; assertion failed: `(left == right)` (left: `1498852938`, right: `554869826`): Unexpected MAGIC_COOKIE: actual=0x5956ae4a, expected=0x2112a442)
    HISTORY:
      [0] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.2.0/src/message.rs:266
      [1] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.2.0/src/message.rs:295 -- message_type=Type { class: Request, method: Method(1) }, message_len=0, transaction_id=TransactionId(0xA24247FFEC0A5D5BA094C0C0)
      [2] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.15/src/combinator.rs:1368
      [3] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.2.0/src/message.rs:430
      [4] at src/turn_message.rs:32
      [5] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.15/src/decode.rs:448
      [6] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/udp.rs:199 -- peer=192.168.0.11:35339
      [7] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/error.rs:19 -- original_error_kind=InvalidInput
      [8] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/share.rs:48
      [9] at src/transport/stun.rs:46
      [10] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.4.0/src/transport/udp.rs:358
      [11] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.4.0/src/transport/udp.rs:193
      [12] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.4.0/src/channel.rs:176
      [13] at /home/a/.cargo/registry/src/github.com-1ecc6299db9ec823/rustun-0.4.0/src/error.rs:61 -- original_error_kind=CodecError
      [14] at src/server/core.rs:425
      [15] at src/server/mod.rs:52
